client/components/domains/register-domain-step/index.jsx

```diff
@@ -441,7 +441,9 @@ const RegisterDomainStep = React.createClass( {
 		const { lastDomainSearched, lastDomainStatus } = this.state,
 			matchesSearchedDomain = ( suggestion ) => ( suggestion.domain_name === lastDomainSearched ),
 			availableDomain = lastDomainStatus === domainAvailability.AVAILABLE && find( this.state.searchResults, matchesSearchedDomain ),
-			onAddMapping = ( domain ) => this.props.onAddMapping( domain, this.state );
+			onAddMapping = ( domain ) => this.props.onAddMapping( domain, this.state ),
+			exactMatchBeforeTld = ( suggestion ) => ( startsWith( suggestion.domain_name, `${ lastDomainSearched }.` ) ),
```

Comment:
What if the user searched for a domain name like `example.com`? `lastDomainSearch` would be `example.com` and this check would not work as it should, right?